src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala

```diff
@@ -223,15 +223,16 @@ class UITestSuite extends BaseGitRepositoryBackedIntegrationTestSuite(SETUP_WITH
     Assert.assertEquals("develop", currentBranchName)
   }
 
-  private def saveThreadDumpToFile(): Unit = {
+  private def saveThreadDumpAndScreenshot(): Unit = {
     val pid = intelliJ.probe.pid()
+    intelliJ.probe.screenshot("exception")
     val threadStackTrace: String = Process("jstack " + pid) !!
-    val file: File = new File("build/thread-dump/thread_dump_" + pid + ".txt")
-    if (file.getParentFile.mkdirs()) {
-      val pw = new PrintWriter(file)
-      pw.write(threadStackTrace)
-      pw.close()
-    }
+    val artifactDirectory = "/tmp/ide-probe/screenshots/uiTest/exception"
```

Comment:
Actually! if some other UI tests were executed before the failing one... I suppose there will be more artifacts in this directory, not just the ones from the failing test, right?

Reply:
you mean more screenshots ? Yes, but only those with  'exception' and 'on-exit ' in name will be probably important to us (but maybe not ?). I can make some regex in config.yaml and copy to `/exception-artifacts` directory only pictures that have  'exception' or 'on-exit ' in name. But maybe some pictures before them give us some answer about exception

Reply:
Hmmm actually keeping the other screenshots **from the same test method** is very welcome... but won't the directory be spammed with screenshots from:
1. other UI tests for previously tested IntelliJ versions?
2. previously executed test methods for the same IntelliJ version as the failing one?

Reply:
yes .. , the only solution I can see is to solve this problem is to clear` /tmp/ide-probe/screenshots` directory as first action in this test method

Reply:
Let's leave as is as for now, I suppose that once VirtusLab/ide-probe#195 is sorted, we'll have the screenshot automatically distributed by test class/test method-name folders

Reply:
> won't the directory be spammed with screenshots from:

Actually... can you show such a sample CircleCI build which has screenshots from preceding test runs? 🤔 I assume their numbering might go well into hundreds if answers to both (1.) and (2.) are indeed YES

Reply:
I need to run such build

Reply:
yes I think it will be overwritten too, but is it a problem for us ? It will be overwritten until exception occurred and in artifacts only screenshots from last failed task will be kept

Reply:
Huh the only problem might be with having e.g. both `42.png` and `42-on-exit.png` (from different IntelliJ versions), or having screenshots with number **past the final number from the last run**, coming from previous runs... this might lead to some confusion when viewing.

Reply:
> append IntelliJ version to IDEPROBE_SCREENSHOTS_DIR to avoid files getting overwritten

Ouch, I forgot this won't work due to https://github.com/VirtusLab/ide-probe/issues/195... maybe append to `IDEPROBE_TEST_SUITE` then?

Reply:
okey so finally add intellij version to `IDEPROBE_TEST_SUITE` and add to artifacts all pictures from tested version ? (if it fails ofc). (Now I am running CI build to check how much approximately screenshots are taken during test suite for one intellij version )

Reply:
>  add intellij version to IDEPROBE_TEST_SUITE

Yes, AFAICS this would make sense

>  add to artifacts all pictures from tested version 

It'll be cumbersome to determine from CI config **on which specific IntelliJ version** the UI tests failed :// so I'd just include tests from all versions (simply `/tmp/ide-probe/screenshots`), including the ones on which the tests did **not** fail. They might actually serve as a benchmark for comparison in a way (to see where the screenshots from successful&failed tests diverged)

Reply:
yes it really make sense. Testing one intellij version produce around 100 pictures - so CircleCi  shouldn't have a problem with that amount (even as the version for testing will be 10 or more)